de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/types/structural/tree/IBeanStructuralTreeTraverserMatcher.java

```diff
@@ -31,5 +31,17 @@
 	 * @param matchingParent the closest parent of the node that also matches the criteria.
 	 *        Can be null if the node does not have a matching parent
 	 */
-	void foundMatch(IBeanStructuralElementInstance treeNode, IBeanStructuralElementInstance matchingParent);
+	void processMatch(IBeanStructuralElementInstance treeNode, IBeanStructuralElementInstance matchingParent);
+	
+	/**
+	 * This method tells if the children of the current node should be processed or not.
+	 * 
+	 * Default behavior is that all children are processed.
+	 * @param treeNode the node of which the children should be processed or not.
+	 * @param isMatching the matching result which the traverser evaluated before.
+	 * @return true in case the children should be processed. false in case not.
+	 */
+	default boolean processChildren(IBeanStructuralElementInstance treeNode, boolean isMatching) {

```

Comment:
Do we really want to pass `isMatching`? We have this information in the matcher, then pass it to the traverser, and then pass it back to the matcher. If this information is relevant, we can just check it with `isMatching` in the matcher. I would suggest we remove it to have a cleaner interface.

Reply:
I see the point with the interface. Still, in CEF is a case where we need to process the isMatching again in the processing method, which would result in a duplicate call and duplicate processing of the same method which is quite inefficient.

Since it is just an input parameter to the method i prefer leaving it in, since it does not need to be considered for the methods implementation if it has no value.